### PR TITLE
pkg/libcose: add RIOT as crypto backend

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR adds RIOT crypto as a potential crypto backend for libcose. Here only `chacha20poly1305` is added, but this puts the base to extend.

As its only a crypto backend and not usable for signing I also split the encrypting part of the test into a generic test easy to extend with other crypto (stolen from libcose)

### Testing procedure

- green ci
- review the random patch change to adapt to latest libcose master (many new stuff lately)

### Related PRs

Waiting for https://github.com/bergzand/libcose/pull/128
Waiting for https://github.com/RIOT-OS/RIOT/pull/18003

### CODE REVIEW

File: pkg/libcose/include/cose/crypto/riot.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * @{
+ *
+ * @file
+ * @brief       Crypto function api for glueing RIOT crypto libraries
+ *
+ * @author      Francisco Molina <francois-xavier.molina@inria.fr>
+ */
+
+#ifndef CRYPTO_RIOT_H
+#define CRYPTO_RIOT_H
```
Comment (benpicco) [inline pkg/libcose/include/cose/crypto/riot.h:22]:
```suggestion
#ifndef COSE_CRYPTO_RIOT_H
#define COSE_CRYPTO_RIOT_H
```

File: pkg/libcose/include/cose/crypto/riot.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ */
+#define HAVE_ALGO_CHACHA20POLY1305
+
+/** @} */
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* CRYPTO_RIOT_H */
```
Comment (benpicco) [inline pkg/libcose/include/cose/crypto/riot.h:41]:
```suggestion
#endif /* COSE_CRYPTO_RIOT_H */
```

File: pkg/libcose/patches/0001-RIOT-Use-RIOT-random_bytes-function-instead-of-rando.patch [hunk 3]
```diff
@@ -1,94 +1,87 @@
...
+ 
+-cose_crypt_rng cose_crypt_get_random = NULL;
++static int _riot_random_bytes(void* arg, unsigned char * buf, size_t len)
++{
++    (void)arg;
++    random_bytes((uint8_t*) buf, len);
++    return 1;
++}
++
++cose_crypt_rng cose_crypt_get_random = _riot_random_bytes;
```
Comment (benpicco) [inline pkg/libcose/patches/0001-RIOT-Use-RIOT-random_bytes-function-instead-of-rando.patch:32]: Why not drop the patch and call `cose_crypt_set_rng()` instead?

Comment (fjmolinas) [reply pkg/libcose/patches/0001-RIOT-Use-RIOT-random_bytes-function-instead-of-rando.patch]: Done, but now depends on a libcose update

Comment (fjmolinas) [reply pkg/libcose/patches/0001-RIOT-Use-RIOT-random_bytes-function-instead-of-rando.patch]: Droped

**Comment by fjmolinas** [comment]
@benpicco last push should have fixed the ci (only blacklisting missing), ok to squash?

**Review by benpicco — APPROVED** [review]

**Comment by fjmolinas** [comment]
All green here @bergzand @benpicco

**Comment by fjmolinas** [comment]
Thanks!

**Pull request merged**